itest/rust/src/object_test.rs

```diff
@@ -762,3 +763,65 @@ pub mod object_test_gd {
         }
     }
 }
+
+// ----------------------------------------------------------------------------------------------------------------------------------------------
+
+#[derive(GodotClass)]
+#[class(init, base=Object)]
+struct DoubleUse {
+    #[base]
+    base: Base<Object>,
+
+    used: atomic::AtomicBool,
```

Comment:
Can we not use `Cell<bool>` here?

We're not dealing with threads as far as I can see.

Reply:
i think so but I just find atomic bools easier  most of the time

Reply:
Apart from the fact that micro-optimizations don't matter in tests: atomics need synchronization (OS support) for thread safety. `std::cell::Cell` on the other hand is a direct wrapper around `UnsafeCell`, and since it's statically guaranteed to be single-threaded, it can optimize its read/writes [to (almost) pointer read/writes](https://doc.rust-lang.org/src/core/cell.rs.html#408-414).

Use the simplest tool for the job. Atomics are confusing in single-threaded contexts 🙂

Reply:
i honestly find atomics less confusing but sure I'll change it

Reply:
> i honestly find atomics less confusing

With the whole `atomic::Ordering::{Release/Acquire}` ceremony? That's all complexity you don't have with `Cell`.

`Cell` is really just `set` + `get`, it cannot possibly get any simpler.

Reply:
if you just use acquire + release (or seqcst everywhere) then yeah it's easy. acquire means to load and release means to store.

it's when you start mixing in relaxed and such it gets confusing. but even then, in single threaded code im pretty sure relaxed is the same as seqcst. but i just use acquire and release/acqrel. in fact i believe Cell<bool> and AtomicBool with relaxed are semantically identical in single threaded code.